File: spring-boot/converter/src/main/java/com/example/converter/entity/Member.java
package com.example.converter.entity;

import com.example.converter.converter.AES256Converter;
import com.example.converter.converter.BooleanToTinyintConverter;
import lombok.*;

import javax.persistence.*;

@Getter
@Builder
@AllArgsConstructor
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Entity
public class Member {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;
    private String name;
    @Convert(converter = AES256Converter.class)
    private String regNo;
    @Convert(converter = BooleanToTinyintConverter.class)
    private boolean deleted;
}

File: spring-boot-kotlin/converter/src/main/kotlin/com/example/converter/member/entity/Member.kt
package com.example.converter.member.entity

import com.example.converter.member.converter.AES256Converter
import com.example.converter.member.converter.BooleanToTinyintConverter
import jakarta.persistence.*

@Entity
data class Member(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private var id: Long?,

    val name: String,

    @Convert(converter = AES256Converter::class)
    val regNo: String,

    @Convert(converter = BooleanToTinyintConverter::class)
    val deleted: Boolean = false
)